nni/algorithms/compression/v2/pytorch/pruning/tools/rl_env/agent.py

```diff
@@ -0,0 +1,232 @@
+        self.critic_optim = Adam(self.critic.parameters(), lr=self.ddpg_params['lr_c'])
+
+        self.hard_update(self.actor_target, self.actor)  # Make sure target is with the same weight
+        self.hard_update(self.critic_target, self.critic)
+
+        # Create replay buffer
+        self.memory = SequentialMemory(limit=self.ddpg_params['rmsize'], window_length=self.ddpg_params['window_length'])
+
+        # Hyper-parameters
+        self.batch_size = self.ddpg_params['bsize']
+        self.tau = self.ddpg_params['tau']
+        self.discount = self.ddpg_params['discount']
+        self.depsilon = 1.0 / self.ddpg_params['epsilon']
+        self.lbound = 0.  # self.ddpg_params['lbound']
+        self.rbound = 1.  # self.ddpg_params['rbound']
+
+        # noise
+        self.init_delta = self.ddpg_params['init_delta']
+        self.delta_decay = self.ddpg_params['delta_decay']
+        self.warmup = self.ddpg_params['warmup']
+
+        self.epsilon = 1.0
+        # self.s_t = None  # Most recent state
+        # self.a_t = None  # Most recent action
+        self.is_training = True
+
+        #
+        if USE_CUDA: self.cuda()
+
+        # moving average baseline
+        self.moving_average = None
+        self.moving_alpha = 0.5  # based on batch, so small
+
+    def update_policy(self):
+        # Sample batch
+        state_batch, action_batch, reward_batch, \
+        next_state_batch, terminal_batch = self.memory.sample_and_split(self.batch_size)
+
+        # normalize the reward
+        batch_mean_reward = np.mean(reward_batch)
+        if self.moving_average is None:
+            self.moving_average = batch_mean_reward
+        else:
+            self.moving_average += self.moving_alpha * (batch_mean_reward - self.moving_average)
+        reward_batch -= self.moving_average
+
+        # Prepare for the target q batch
+        with torch.no_grad():
+            next_q_values = self.critic_target([
+                to_tensor(next_state_batch),
+                self.actor_target(to_tensor(next_state_batch)),
+            ])
+
+        target_q_batch = to_tensor(reward_batch) + \
+                         self.discount * to_tensor(terminal_batch.astype(np.float)) * next_q_values
+
+        # Critic update
+        self.critic.zero_grad()
+
+        q_batch = self.critic([to_tensor(state_batch), to_tensor(action_batch)])
+
+        value_loss = criterion(q_batch, target_q_batch)
+        value_loss.backward()
+        self.critic_optim.step()
+
+        # Actor update
+        self.actor.zero_grad()
+
+        policy_loss = -self.critic([ # pylint: disable=all
+            to_tensor(state_batch),
+            self.actor(to_tensor(state_batch))
+        ])
+
+        policy_loss = policy_loss.mean()
+        policy_loss.backward()
+        self.actor_optim.step()
+
+        # Target update
+        self.soft_update(self.actor_target, self.actor)
+        self.soft_update(self.critic_target, self.critic)
+
+    def eval(self):
```

Comment:
Why doesn't `DDPG` inherit `nn.Module` so that `eval`, `cuda` are no longer required.

Reply:
This file is migrated from v1 and there's no many changes. But that sounds good, I would try to optimize it.